### ISSUE TITLE
Deferred errors makes requirements checker less useful

### ISSUE THREAD

**Issue by cdsmith** [google/codeworld#865]
**Shared Program**: https://code.world/#PYurNtPA2Z7wBP-5D1xq0rA

**Description**

It used to be that the requirements checker didn't run unless your program worked.  This was because of https://github.com/google/codeworld/commit/b4e36e443789ae2bf193d2c36de3e369fb295b0e and was done deliberately because it was confusing to students that the requirements checker marks their project as correct even though the code doesn't compile.

Unfortunately, now that we defer many errors until runtime, we're back in a state where the requirements checker succeeds when there are build errors.  I see a few possibilities here.

1. Add some logic to detect deferred errors at build time and suppress the requirement checker.  However, this has the disadvantage that if the deferred error is truly unnecessary (e.g., in an unreferenced variable), the requirements checker doesn't appear to run even when the program appears to work.

2. Extend the requirements language (or just add a silent requirement by default) to require that there are no errors, and display an actual requirements dialog, but just visibly fail because there are errors.

3. Push forward to doing runtime verification, too, and then the requirement checker would fail during runtime verification.  But this is a very speculative project, and it's a long way off.

4. Show the requirement checker UI only after the program doesn't crash for a couple seconds.  Kludgy, but it doesn't sound too hard.

5. Just disable deferred errors if there are requirements.  I'd prefer not to...

Currently, I favor option 4 for the solution.

**Comment by devanshbatra04** [comment]
Can I take this up. Would love to start contributing :smile:

**Comment by cdsmith** [comment]
@devanshbatra04 yes, that would be fine with me.  This should be a good bug to get started with.  Let me know if you need help finding the code to change.
